app/femr/ui/controllers/admin/InventoryController.java

```diff
@@ -314,17 +314,35 @@ public Result existingPost(Integer tripId) {
                     //create a non-concept MedicationItem from the concept MedicationItem
                     conceptMedicationItem = conceptMedicationServiceResponse.getResponseObject();
                     medicationItemServiceResponse = medicationService.createMedication(conceptMedicationItem.getName(), conceptMedicationItem.getForm(), conceptMedicationItem.getActiveIngredients());
-
                     if (medicationItemServiceResponse.hasErrors()) {
 
                         return internalServerError();
                     }else{
 
-                        ServiceResponse<MedicationItem> setQuantityServiceResponse = inventoryService.setQuantityTotal(medicationItemServiceResponse.getResponseObject().getId(), tripId, 0);
+                        //If it exists in the formulary, delete or un-delete the medication from the formulary
+                        inventoryService.deleteInventoryMedication(medicationItemServiceResponse.getResponseObject().getId(), tripId);
+
+                        //Find the medication so we can find it's initial quantity, and current quantity
+                        ExpressionList<MedicationInventory> medicationInventoryExpressionList = QueryProvider.getMedicationInventoryQuery()
```

Comment:
Are you able to call the service layer instead of re-writing this?:

https://github.com/FEMR/femr/blob/master/app/femr/business/services/system/InventoryService.java#L117